src/Processors/Formats/IRowInputFormat.cpp

```diff
@@ -98,36 +101,53 @@ Chunk IRowInputFormat::generate()
                 if (!isParseError(e.code()))
                     throw;
 
-                if (params.allow_errors_num == 0 && params.allow_errors_ratio == 0)
-                    throw;
+                if (!append_error_column)
```

Comment:
Maybe we can just set `allow_errors_ratio` to 1 and add column if `append_error_column`, but this branch execute as earlier.

Reply:
Why not just catching exception on the caller side?

Reply:
> Why not just catching exception on the caller side?
We should just handle the current row message, and collect the message, then process the next one.
It's better not to break the loop.

Reply:
> Maybe we can just set `allow_errors_ratio` to 1 and add column if `append_error_column`, but this branch execute as earlier.

Yes, I'll fix it.

Reply:
I don't like current approach.

It will not handle errors from readPrefix / readSuffix. And will work only for IRowInputFormat based formats. 

We process single message at a time.
You can put try catch in loop (I've added comment about the place we should catch it)

Reply:
Also that approach 
1) pollute clean interface of IRowInputFormat
2) may affect parsers performance for non-kafka case. 

For me approach with try catch in Kafka sounds much more clean.

Reply:
> Also that approach
> 
> 1. pollute clean interface of IRowInputFormat
> 2. introduce extra states for IRowInputFormat
> 3. may affect parsers performance for non-kafka case.
> 
> For me approach with try catch in Kafka sounds much more clean.

Sure. I agree with you. I'll change the current approach.